src/manifolds/Grassmann.jl

```diff
@@ -149,7 +149,7 @@ function distance(M::Grassmann, p, q)
     p ≈ q && return zero(real(eltype(p)))
     a = svd(p' * q).S
     a[a .> 1] .= 1
-    return sqrt(sum((acos.(a)) .^ 2))
+    return sqrt(sum((real(acos.(a))) .^ 2))
```

Comment:
The above inequality check suggests that `a` contains reals (inequalities fail for complex). `acos` will then only produce reals, so why is the call to `real` needed?
Also, you could use `sum(x -> abs2(real(acos(x))), a)` to avoid allocating `acos.(a)`.

Reply:
That's merely for numerical precision to add the real, I think.

Reply:
The line `a[a .> 1] .= 1` will error if `eltype(a) <: Complex`:

```julia
julia> (1.0 + 0im) < 3
ERROR: MethodError: no method matching isless(::Complex{Float64}, ::Int64)
Closest candidates are:
  isless(::Missing, ::Any) at missing.jl:87
  isless(::AbstractFloat, ::Real) at operators.jl:167
  isless(::Real, ::Real) at operators.jl:355
```
How does `real(x::Real)` add numerical precision? Isn't that a no-op?

Reply:
For the first: good observation, I will look how to fix that.

For the second: Until now we mostly thought of this lines in the real case, yes. The real stems from here https://www.manopt.org/reference/manopt/manifolds/grassmann/grassmanncomplexfactory.html - line 63, if you have rounding errors and end up with complex numbers I think.

Reply:
maybe it would then be better to do a `real(a)`?

Reply:
Hm, perhaps matlab handles `acos(1.1)` by extending to the complex plane with `acos(1.1+0im) => 0.0 - 0.4435682543851153im`? Since the code does not clamp `cos_princ_angle`, then one would need to call `real` on the output, which is effectively the same as our clamping.

That being said, I am pretty sure singular values are always real numbers, so calling `real(a)` should not be necessary.

Reply:
Ok, I left it out again.